In stage 0 glibc configure tries to call the build system gcc with -mabi=32,
causing test failures. Also set _MIPS_SZPTR for 32bit builds since it is not
defined elsewhere.

See http://sourceware.org/ml/crossgcc/2005-07/msg00036.html and
http://sourceware.org/ml/crossgcc/2008-01/msg00007.html

--- glibc-2.7/ports/sysdeps/mips/mips32/Makefile-orig	2003-03-29 09:15:28.000000000 +0100
+++ glibc-2.7/ports/sysdeps/mips/mips32/Makefile	2008-10-06 03:29:36.000000000 +0200
@@ -1,3 +1,3 @@
 ifeq ($(filter -mabi=32,$(CC)),)
-CC += -mabi=32
+CC += -D"_MIPS_SZPTR=32"
 endif
